frontend/src/metabase/query_builder/components/QuestionActions.tsx

```diff
@@ -215,6 +246,24 @@ const QuestionActions = ({
     });
   }
 
+  const handleFileUpload = (event: ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (file && question._card.based_on_upload) {
+      uploadFile({
+        file,
+        tableId: question._card.based_on_upload,
+      })(dispatch);
+
+      // reset the file input so that subsequent uploads of the same file trigger the change handler
+      const fileInput = document.getElementById(
```

Comment:
I think we would generally use a `ref` for this rather than `document.getElementById`. Will that work here?